### ISSUE TITLE
editor.accessibilityPageSize setting  is messed up

### ISSUE THREAD

**Issue by snufas** [microsoft/vscode#166381]
Type: <b>Bug</b>

@isidorn
When I set "editor.accessibilityPageSize": 1
screenreader no longer speaks  when trying to highlight text line by line using shift  + down arrow keys, it works quite fine using left and right arrow keys but not when using up- or down.
It reads the 1st selected line but if you continuously press down shift plus down, it doesn't read the selection anymore.
Also it slows down the screenreader response.

when "editor.accessibilityPageSize": 1  is enabled it makes magnifier  somewhat working, without it the magnifier does not work properly  at all when following the edit cursor...

The issue described here: Windows Magnifier not following text caret when accessibility support is enabled #105558

so in short:
editor.accessibilityPageSize setting as I mentioned on the above-mentioned issue, doesn't fully fix the magnifier tracking issue, but also breaks the code selection for screenreader users.
Also it definitely slows down the screen reader response when navigating the code...
The issue is also replicable on the nightly build...

VS Code version: Code 1.73.1 (6261075646f055b99068d3688932416f2346dd3b, 2022-11-09T04:27:29.066Z)
OS version: Windows_NT x64 10.0.19044
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-5930K CPU @ 3.50GHz (12 x 3498)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|undefined|
|Memory (System)|31.91GB (22.79GB free)|
|Process Argv|--crash-reporter-id 7c280da0-24a3-4763-9e9f-4345e9e5c40f|
|Screen Reader|yes|
|VM|50%|
</details><details><summary>Extensions (25)</summary>

Extension|Author (truncated)|Version
---|---|---
blackbox|Bla|0.4.2
path-intellisense|chr|2.8.1
vscode-myssql-client2|cwe|4.3.1
denigma|den|0.0.14
docs-markdown|doc|0.2.116
gitlens|eam|13.0.4
prettier-vscode|esb|9.9.0
paste-from-screenshot|jam|0.0.1
vscode-docker|ms-|1.22.2
isort|ms-|2022.6.0
python|ms-|2022.18.2
vscode-pylance|ms-|2022.11.20
jupyter|ms-|2022.9.1202862440
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.12
vscode-jupyter-cell-tags|ms-|0.1.6
vscode-jupyter-slideshow|ms-|0.1.5
remote-containers|ms-|0.262.3
remote-wsl|ms-|0.72.0
vsliveshare|ms-|1.0.5762
text-power-tools|qcz|1.38.0
multi-command|ryu|1.6.0
code-spell-checker|str|2.11.0
intellicode-api-usage-examples|Vis|0.2.6
vscodeintellicode|Vis|1.2.29

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256860
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30410667
cppdebug:30492333
vsclangdc:30486549
c4g48928:30535728
dsvsc012:30540252
azure-dev_surveyone:30548225
pyindex848:30577860
nodejswelcome1cf:30587006
fc301958:30595537
282f8724:30602487
gswce2:30605431
3d0df643:30608728
dbltrim-noruby:30604474
89544117:30607850

```

</details>

**Comment by meganrogge** [comment]
Do you have word wrapping enabled?

**Comment by snufas** [comment]
@meganrogge 
Same problem with or without word wrap  enabled 
Acording to : October 2022 (version 1.73) update notes 
Word wrapping in screen reader mode
Word wrapping had been disabled when running VS Code in screen reader mode to prevent a degraded screen reader experience. This has been fixed and can be enabled via editor.wordWrap.

Basically, when I select 1st line using shift + down arrow screen reader reads  that line, when I press shift + down arrow again to highlight the 2nd line, screenreader says
"..."
And after that  there is no speech output anymore.

I wouldn't use editor.accessibilityPageSize setting if I wouldn't need a magnifier, but I do, particularly if I want to investigate more complicated code visually.
It's a lose/lose situation right now. One thing breaks another thing. The magnifier issue with vs code exists for very long time...
Accessibility on vs code is top-notch as it is, but we need some extra fixing, to make it even better.
Magnifier + screenreader combination is uncommon I guess, because it usually is one or another, but for visually impaired folks like me both are very useful and are used in combination...

**Comment by justas2481** [comment]
I can confirm that problem persist in my setup if I turn on magnifier as well. As I understand it, the biggest trouble here is the requirement to set editor.accessibilityPageSize to 1 so to be able to use Magnifier together with screen reader. Since I'm totally blind, I use screen reader only and editor.accessibilityPageSize is set to 500 by default so I see no trouble whatsoever, but in case of being able to read some parts of the code visually and do the rest of the work without the eyesite, that's troublesome.
One workaround for now is to use ctrl+k ctrl+b to start / begin the selection, then move cursor to the end and use ctrl+k ctrl+k to select part of the code so to avoid shift+down arrow all together.

**Comment by meganrogge** [comment]
I believe @isidorn suggested the workaround is to set accessibility page size to 2. 

That shouldn't be needed soon when the fix is in for https://github.com/microsoft/vscode/issues/105558 shortly, so let's track this there.
